core/extrinsics_submission_service/extrinsic_submission_service.hpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_SERVICE_EXTRINSICS_SUBMISSION_SERVICE_HPP
+#define KAGOME_CORE_SERVICE_EXTRINSICS_SUBMISSION_SERVICE_HPP
+
+#include <vector>
+
+#include <jsonrpc-lean/server.h>
+#include <boost/asio/io_context.hpp>
+#include <boost/signals2/signal.hpp>
+#include "extrinsics_submission_service/extrinsic_submission_proxy.hpp"
+#include "extrinsics_submission_service/json_transport.hpp"
+#include "extrinsics_submission_service/network_address.hpp"
+
+namespace kagome::service {
+
+  /**
+   * @brief extrinsic submission service implementation
+   */
+  class ExtrinsicSubmissionService {
+    using SignalType = void(std::string_view);
+    template <class T>
+    using sptr = std::shared_ptr<T>;
+
+   public:
+    /**
+     * @brief service configuration
+     */
+    struct Configuration {
+      NetworkAddress address{};
+    };
+    /**
+     * @brief constructor
+     * @param context io_context reference
+     * @param config server configuration
+     * @param api_proxy extrinsic submission api proxy reference
+     */
+    ExtrinsicSubmissionService(Configuration configuration,
+                               std::shared_ptr<JsonTransport> transport,
```

Comment:
Why not
```suggestion
                               sptr<JsonTransport> transport,
```
if it's defined?

Reply:
Using it in interface is a bad idea. Everyone know std::shared_ptr, but who knows what is sptr?